pydrive2/files.py

```diff
@@ -351,24 +351,38 @@ def download(fd, request):
             # metadata["mimeType"].startswith("application/vnd.google-apps.").
             # But that would first require a slow call to FetchMetadata().
             # We prefer to try-except for speed.
-            try:
-                download(fd, files.get_media(fileId=file_id))
-            except errors.HttpError as error:
-                exc = ApiRequestError(error)
-                if (
-                    exc.error["code"] != 403
-                    or exc.GetField("reason") != "fileNotDownloadable"
-                ):
-                    raise exc
-                mimetype = mimetype or "text/plain"
-                fd.seek(0)  # just in case `download()` modified `fd`
+
+            # If mimeType is provided we should use that with export_media
+            # immmediately to save an extra API call that the caller knows is
+            # like to fail, otherwise try get_media then export_media
+
+            if mimetype:
                 try:
                     download(
                         fd,
                         files.export_media(fileId=file_id, mimeType=mimetype),
                     )
                 except errors.HttpError as error:
                     raise ApiRequestError(error)
+            else:
+                try:
+                    download(fd, files.get_media(fileId=file_id))
+                except errors.HttpError as error:
+                    exc = ApiRequestError(error)
+                    if (
+                        exc.error["code"] != 403
+                        or exc.GetField("reason") != "fileNotDownloadable"
+                    ):
+                        raise exc
+                    mimetype = "text/plain"
+                    fd.seek(0)  # just in case `download()` modified `fd`
+                    try:
+                        download(
+                            fd,
+                            files.export_media(fileId=file_id, mimeType=mimetype),
+                        )
+                    except errors.HttpError as error:
+                        raise ApiRequestError(error)
```

Comment:
falling back to downloading `text/plain` wasn't the old behaviour...

```suggestion
                    mimetype = "text/plain"
                    fd.seek(0)  # just in case `download()` modified `fd`
```

Reply:
Hmm, am I'm missing something - it looks like if `mimetype` was not provided in the previous version we were using "text/plain" indeed?

the only behavior change here is that we always now do `export_media` if `mimetype` is provided. How valid is that? is it specified somewhere that `mimetype` _always_ means that file _must_ be downloaded with `export_media`?

Reply:
> Hmm, am I'm missing something - it looks like if `mimetype` was not provided in the previous version we were using "text/plain" indeed?

no, the previous version just ignored `mimetype` for the purposes of downloading. It only ever used `mimetype` for the BOM removal logic.

> the only behavior change here is that we always now do `export_media` if `mimetype` is provided. How valid is that? is it specified somewhere that `mimetype` _always_ means that file _must_ be downloaded with `export_media`?

hmm yes not sure about this either - related to https://github.com/iterative/PyDrive2/pull/73#discussion_r546240791